### PR TITLE
Mediapipe face tracking extension

### PR DESCRIPTION
**Pull request by mgrewe**
Hi all, 

this extension is meant to bring webcam-based face tracking into jsPsych. It can be used to record high-level posture movements like translation and rotation (estimated in 3D space), as well as the locations of a number of facial landmarks. This way, (1) facial movements can be measured without the need to record and collect sensitive facial images (GDPR conformity), and (2) interactive content based on the participant's facial movements (like avatars mirroring the participants) can be implemented.

This is a first stub only, and I'm currently struggling with a correct integration of the Mediapipe library as a dependency.

Cheers, Martin

### CODE REVIEW

**Comment by bjoluc** [comment]
Tricky situation indeed! The problem is that the `@mediapipe/face_mesh` package pretends to provide an ES module (both TypeScript-wise and via the `module` key in `package.json`) while it doesn't provide one (https://github.com/google/mediapipe/issues/3167). Instead, it attaches the `FaceMesh` class directly to the global object. You can still bundle FaceMesh by using a plain `import "@mediapipe/face_mesh";`. I've committed a hack in global.d.ts to make TypeScript aware of the global `FaceMesh` class. Hope this helps as a starting point!

File: packages/extension-mediapipe-face_mesh/package.json [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    "test": "jest",
+    "test:watch": "npm test -- --watch",
+    "tsc": "tsc",
+    "build": "rollup --config",
+    "build:watch": "npm run build -- --watch"
+  },
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/jspsych/jspsych-contrib.git",
+    "directory": "packages/extension-template-ts"
```
Comment (bjoluc) [inline packages/extension-mediapipe-face_mesh/package.json:29]: Needs an update

**Comment by mgrewe** [comment]
Great! Thanks a lot for your support. Going to work on this and provide a new commit soon.

**Comment by mgrewe** [comment]
Hi @bjoluc,
I committed a working version of the extension. Would be great if you could review.
Thanks, Martin

**Review by bjoluc — CHANGES_REQUESTED** [review]
LGTM! I don't mind the copy-pasted type defs; don't know if there'a any better way to make all of them global at once. Would you mind creating a README.md using readme-template.md and adding this extension to the table in the root README.md?

**Comment by mgrewe** [comment]
Thanks for the support, I hope this is ready for the merge then.

**Review by bjoluc — APPROVED** [review]
Thanks for your contribution @mgrewe! :tada:

**Comment by mgrewe** [comment]
Thanks for the review, and especially the joint forces! Looking forward to my next contribution ;-)

**Pull request merged**